elytron/src/main/java/org/wildfly/extension/elytron/ClassLoadingAttributeDefinitions.java

```diff
@@ -22,6 +22,7 @@ class ClassLoadingAttributeDefinitions {
     static final SimpleAttributeDefinition MODULE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.MODULE, ModelType.STRING, true)
         .setAttributeGroup(ElytronDescriptionConstants.CLASS_LOADING)
         .setAllowExpression(false)
+        .setCorrector(ModuleIdentifierUtil.MODULE_NAME_CORRECTOR)
```

Comment:
Since the module will be provided now in a canonicalized form, apparently, we can now remove the canonicalization request done at https://github.com/wildfly/wildfly-core/blob/481cbde8dc6ac6a2a8b29353f6b85879db54a09d/elytron/src/main/java/org/wildfly/extension/elytron/ClassLoadingAttributeDefinitions.java#L46


The only one I have in doubt is the invocation of that method from https://github.com/wildfly/wildfly-core/blob/bb0806bacc370a49141f0c9cd9be4fb12066a0bb/elytron/src/main/java/org/wildfly/extension/elytron/JaasCustomSecurityRealmWrapper.java#L63

@darranl What's your feeling about this? I guess the ultimate goal here by moving the module name corrector to the attributes is also to avoid parsing to the canonize form twice.

Reply:
Unless I am misreading this comment this sounds like a breaking change we can not make to the management model.

Is the call to parseCanonicalModuleIdentifier more permissive than ModuleIdentifierUtil.MODULE_NAME_CORRECTOR?

Reply:
The idea has always been to place this sort of conversions/adjustments as close as possible to the external inputs, hence on the attribute definitions via `ModuleIdentifierUtil.MODULE_NAME_CORRECTOR`, however, for this [JaasCustomSecurityRealmWrapper.java](https://github.com/wildfly/wildfly-core/blob/bb0806bacc370a49141f0c9cd9be4fb12066a0bb/elytron/src/main/java/org/wildfly/extension/elytron/JaasCustomSecurityRealmWrapper.java) class, which is indeed deprecated, I'm not sure what the external input is for this `configuration` object